src/frontends/lean/structure_cmd.cpp

```diff
@@ -654,7 +654,8 @@ struct structure_cmd_fn {
                     expr base_obj = *mk_base_projections(m_env, parent_name, base_S_name, field);
                     level_param_names lparams; unsigned nparams; inductive::intro_rule intro;
                     std::tie(lparams, nparams, intro) = get_parent_info(base_S_name);
-                    expr type = instantiate_univ_params(m_p.env().get(full_fname).get_type(), lparams, const_levels(parent_fn));
+                    levels meta_ls = map2<level>(lparams, [&](name const &) { return m_ctx.mk_univ_metavar_decl(); });
+                    expr type = instantiate_univ_params(m_p.env().get(full_fname).get_type(), lparams, meta_ls);
```

Comment:
@kha We should discuss this issue next week during our dev meeting. It is not clear to me what the problem is.
Before you refactored this module, I would create a fake `Pi` (or `lambda`, I don't remember), and elaborate everything in as a single call to the elaborator. I'm assuming this approach does not work (i.e., it fails with some delayed abstractions as your reported above). Is this assumption correct?